### PR TITLE
TVX-356 New ad-hoc command to set and get additional properties on a user

### PR DESCRIPTION
**Pull request by viv**
This subsumes and replaces #8

### CODE REVIEW

File: src/java/org/jivesoftware/openfire/commands/admin/user/SetUserAdditionalProperty.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+        Element note = command.addElement("note");
+
+        String accountJid;
+        String propertyKey;
+        String propertyValue;
+
+        try {
+            accountJid = data.get("accountjid").get(0);
+            propertyKey = data.get("key").get(0);
```
Comment (viv) [inline src/java/org/jivesoftware/openfire/commands/admin/user/SetUserAdditionalProperty.java]: Check for empty JID (to avoid NPE) and key (to avoid setting the empty property key).

Add tests to confirm empty key can't be set.

**Comment by latentflip** [comment]
If the other's are happy I'm happy as far as code is concerned.

I've just kicked off a build to test in action with the UI stuff will feedback when that's done.

**Review by garethf — APPROVED** [review]

**Review by latentflip — APPROVED** [review]
I am happy with this :+1: have done a get+set via client for both system and user properties and all looks good to me.
